dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/MethodSignature.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+import com.google.common.base.MoreObjects;
+import java.util.Objects;
+
+/**
+ * A representation of a tuple of method name and descriptor (type information) in a Method entry
+ * in a class file. The descriptor part helps to distinguish method entries when method overloading
+ * is used
+ */
+public class MethodSignature {
+  private String methodName;
+  private String descriptor;
+
+  public MethodSignature(String methodName, String descriptor) {
```

Comment:
I don't think "descriptor" is quite right. Maybe "parameterListDescriptor", because this is just the parameter list, it doesn't describe the whole method signature.

Reply:
It includes return types too e.g., "(Lcom/google/protobuf/Message;)Lio/grpc/MethodDescriptor$Marshaller;"

![dimtegbydsn](https://user-images.githubusercontent.com/28604/46555592-c84c2480-c8b1-11e8-9db9-1eafe15a46bd.png)

Reply:
Oh thanks! I learned something :-)